sdk/trace/span.go

```diff
@@ -444,24 +421,17 @@ func truncate(limit int, s string) string {
 // If this method is called while panicking an error event is added to the
 // Span before ending it and the panic is continued.
 func (s *recordingSpan) End(options ...trace.SpanEndOption) {
-	// Do not start by checking if the span is being recorded which requires
-	// acquiring a lock. Make a minimal check that the span is not nil.
 	if s == nil {
 		return
 	}
 
-	// Store the end time as soon as possible to avoid artificially increasing
-	// the span's duration in case some operation below takes a while.
-	et := monotonicEndTime(s.startTime)
-
-	// Lock the span now that we have an end time and see if we need to do any more processing.
-	s.mu.Lock()
-	if !s.isRecording() {
-		s.mu.Unlock()
+	config := trace.NewSpanEndConfig(options...)
```

Comment:
I think building the EndConfig before setting the time is okay since it's only setting some fields on a struct on the stack. We could move the config below, but it would make Span.EndTime racy. A caller could observe both the time.Now value and the config value.